src/main/java/eu/europeana/metis/sandbox/util/ZipFileReader.java

```diff
@@ -0,0 +1,78 @@
+package eu.europeana.metis.sandbox.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class provides the functionality of reading zip files.
+ * <br /><br />
+ * TODO search ZipFileReader class in project metis-common and add a method that also returns
+ *   records as List<byte[]> then replace this class with that one
+ */
+public class ZipFileReader {
```

Comment:
This is a copy of a class in metis-common project, but it returns a list of byte array instead of a list of strings. I think a method that returns this could be added there and then we can replace this class, but for now I modify it and put it here

Reply:
Let's do this the proper way. Let's create that method in metis common!

Reply:
Yes I thought of that, but then I will have to require snapshot versions of metis libraries instead of current release, which I could do, but I thought it was better for the sandbox to stay as a client of metis libraries as they are now (release 1.8.0 I think).

I see you evolve metis libraries in sync with ecloud code; so probably in the future sandbox releases could be in sync with metis complete environment releases. For the first production deployment I did not think of depending on metis deployment but maybe that is wrong.

But if we are not making the code available in sync then even if I go and add the method to metis-common I will only be able to use it if looking at a snapshot.

I am open to suggestions. Maybe I am not understanding something on the process of release.
What do you think?

Reply:
[MET-2680](https://europeana.atlassian.net/browse/MET-2680)